src/content/docs/guides/Beginners/Your First Shader/0_intro.mdx

```diff
@@ -0,0 +1,57 @@
+:::
+
+You will need:
+- A suitable text editor for modifying shader code. One popular choice is [Visual Studio Code](https://code.visualstudio.com/), but any program which can edit text files is sufficient. A simpler but popular editor is [Notepad++](https://notepad-plus-plus.org/).
+- A computer capable of running OpenGL 3.3. If your computer was manufactured in the last 15 years it likely supports OpenGL 3.3. Note that whilst OpenGL works on MacOS, it is deprecated, and some things may not work correctly.
+- A willingness to learn. Shaders are hard, and you can't just pick them up overnight. Do not expect to complete this tutorial and become the next Sonic Ether.
+- An instance of Minecraft with Iris, Optifine, or Oculus installed. Since this tutorial is in the Iris documentation, it is assumed you are using Iris.
+
+It is recommended that you use a Minecraft launcher which supports viewing the logs in realtime. One such choice is [Prism Launcher](https://prismlauncher.org/).
+In Iris, it is also useful to enable debug mode. You can do this by pressing <kbd>Ctrl</kbd>+<kbd>D</kbd> in the shader selection screen.
+
+
+## OpenGL and GLSL
+
+Old versions of Minecraft used an ancient version of OpenGL - OpenGL 2.0. Most older shader packs, for this reason, were written using GLSL version 120. However, since Minecraft 1.17, Minecraft was updated to use version 3.2, which targets version 150. However, shader loader mods like Iris and Optifine allow the use of any version of OpenGL/GLSL the user's hardware supports. Since OpenGL 3.3 requires 'DX10 Class' hardware, any computer released in the last decade should support it, so this is what we will be using. OpenGL 3.3 uses GLSL version 330.
+
+For the next part it is important to note that since most of the game's development was done on OpenGL 2.0, the rendering pipeline the game uses does not take advantage of modern hardware features.
+
+## Rendering the blocks
+
+Now with that out of the way, we can focus on how Minecraft actually does it's rendering. Minecraft is a voxel game, and therefore it does not follow the normal style of rendering that is present in most games.First of all, Minecraft has to render a large amount of blocks, which could be different types of blocks. Rendering each block as it's own draw call is a really bad idea for performance. Instancing could work, but it has it's own limitations, besides not being present in ancient version of OpenGL at all. Instead, what Minecraft does it it batches vertices into chunks, so that each chunk becomes it's own draw call. To texture each block, Minecraft uses a texture atlas.
```

Comment:
```suggestion
Now with that out of the way, we can focus on how Minecraft actually does it's rendering. Minecraft is a voxel game, and therefore it does not follow the normal style of rendering that is present in most games. First of all, Minecraft has to render a large amount of blocks, which could be different types of blocks. Rendering each block as it's own draw call is a really bad idea for performance. Instancing could work, but it has it's own limitations, besides not being present in ancient version of OpenGL at all. Instead, what Minecraft does it it batches vertices into chunks, so that each chunk becomes it's own draw call. To texture each block, Minecraft uses a texture atlas.
```